js/core/option_manager.js

```diff
@@ -1,26 +1,27 @@
 import coreDataUtils from "./utils/data";
 import { equals } from "./utils/comparator";
 import typeUtils from "./utils/type";
-import createCallBack from "./utils/callbacks";
+import { noop } from "./utils/common";
 import { extend } from "./utils/extend";
 
+let cachedDeprecateNames = [];
+let cachedGetters = {};
+let cachedSetters = {};
+
 export class OptionManager {
     constructor(options, optionsByReference, deprecatedOptions) {
         this._options = options;
         this._optionsByReference = optionsByReference;
         this._deprecatedOptions = deprecatedOptions;
-        this._changingCallbacks = createCallBack({ syncStrategy: true });
-        this._changedCallbacks = createCallBack({ syncStrategy: true });
-        this._deprecatedCallbacks = createCallBack({ syncStrategy: true });
-        this._cachedDeprecateNames = [];
-        this.cachedGetters = {};
-        this.cachedSetters = {};
+        this._changingCallbacks;
```

Comment:
```suggestion
        this._changingCallback;
```
and what about this:
```suggestion
        this._changingCallback = undefined;
```

Reply:
Same for `_changedCallbacks` and `_deprecatedCallbacks`